src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,624 @@
+    {
+        var buffer = new byte[1024];
+        using ({|#2:HashAlgorithm sha256 = new SHA256Managed()|})
+        {
+            byte[] digest = {|#0:sha256.ComputeHash({|#1:buffer|})|};
+        }
+    }
+}
+";
+
+            string csFix = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        byte[] digest = SHA256.HashData(buffer);
+    }
+}
+";
+            await TestCS(
+                csInput,
+                csFix,
+                VerifyCS.Diagnostic(PreferHashDataOverComputeHashAnalyzer.StringRule)
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1)
+                .WithLocation(2));
+        }
+
+        [Fact]
+        public async Task BasicObjectCreationUsingBlockCase()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        {|#2:Using sha256 As New SHA256Managed()|}
+            Dim digest As Byte() = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim digest As Byte() = SHA256.HashData(buffer)
+    End Sub
+End Class
+";
+            await TestVB(
+                vbInput,
+                vbFix,
+                VerifyCS.Diagnostic(PreferHashDataOverComputeHashAnalyzer.StringRule)
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1)
+                .WithLocation(2));
+
+        }
+
+        [Fact]
+        public async Task BasicCreateHelperUsingBlockCase()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        {|#2:Using sha256 As SHA256 = SHA256.Create()|}
+            Dim digest As Byte() = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim digest As Byte() = SHA256.HashData(buffer)
+    End Sub
+End Class
+";
+            await TestVB(
+                vbInput,
+                vbFix,
+                VerifyCS.Diagnostic(PreferHashDataOverComputeHashAnalyzer.StringRule)
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1)
+                .WithLocation(2));
+
+        }
+
```

Comment:
I don't see any negative cases: tests where the analyzer would detect some of the items, but not all of them, and offer no fix. Can you please add those?

Reply:
I also don't see any test cases where there is more code inside the body of the `using`. That code needs to be preserved.